internal/controller/sync_test.go

```diff
@@ -755,7 +935,13 @@ func TestRoute_buildNextCR(t *testing.T) {
 
 			// check the CSR
 			if tt.wantCSR != nil {
-				csr, err := x509.CreateCertificateRequest(rand.Reader, tt.wantCSR, rsaKey)
+				var privateKey any
+				if tt.wantCSR.PublicKeyAlgorithm == x509.ECDSA {
+					privateKey = ecdsaKey
+				} else if tt.wantCSR.PublicKeyAlgorithm == x509.RSA {
+					privateKey = rsaKey
+				}
+				csr, err := x509.CreateCertificateRequest(rand.Reader, tt.wantCSR, privateKey)
```

Comment:
It was working well with the RSA signature algorithm, but comparing the two signatures will fail with ECDSA since the signatures depend on a random number. I imagine that (for the ECDSA test specifically) we can simply check that the "rest" of the CSR is the same without checking the signature

Reply:
Thanks for the tip @maelvls! It does seem like that is the cause of that test failing. I'll try to update that test accordingly

Reply:
I think I updated the test to cover the ECDSA use case and the rest of the PR is ready for review whenever you have time @maelvls, thanks again for the suggestion 😄